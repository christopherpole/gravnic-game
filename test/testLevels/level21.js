import {
  MOVE_NONE,
  MOVE_UP,
  MOVE_RIGHT,
  ENTITIES,
  levelIsComplete,
  makeMoves,
} from '../../src/index';

describe('Level 21', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.LAVA, id: 1 }, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 2 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 3 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 4 }, movableEntity: null },
      ],
      [
        {
          staticEntity: { entityId: ENTITIES.FLOOR, id: 6 },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 5 },
        },
        {
          staticEntity: { entityId: ENTITIES.GRAVITY_CHANGER, direction: MOVE_NONE, id: 7 },
          movableEntity: null,
        },
        {
          staticEntity: { entityId: ENTITIES.GRAVITY_CHANGER, direction: MOVE_NONE, id: 8 },
          movableEntity: null,
        },
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 9 }, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR, id: 11 },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 10 },
        },
        { staticEntity: { entityId: ENTITIES.FLOOR, id: 12 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.LAVA, id: 13 }, movableEntity: null },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_RIGHT, MOVE_UP, MOVE_RIGHT]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
