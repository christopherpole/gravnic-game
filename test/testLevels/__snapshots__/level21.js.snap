// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 21 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 5,
          },
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 10,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 5,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 10,
          },
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 5,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 10,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 5,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 10,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 5,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "fading": true,
            "id": 10,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 1,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 7,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "direction": "MOVE_NONE",
            "entityId": "GRAVITY_CHANGER",
            "id": 8,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 13,
          },
        },
      ],
    ],
  ],
]
`;
